### ISSUE TITLE
"yaafelib" could not found, and Torch not compiled with CUDA enabled. 

### ISSUE THREAD

**Issue by orasanen** [MarvinLvn/voice-type-classifier#1]
I'm getting the following error on my OSX machine (that has no NVidia card and hence no CUDA): 
```
(pyannote) ws-12129-imac:voice_type_classifier rasaneno$ sh apply.sh /Users/rasaneno/speechdb/demofiles/
apply.sh: line 5: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
Found wav files.
Creating config for pyannote.
apply.sh: line 41: soxi: command not found
Done creating config for pyannote.
Extracting KCHI
awk: can't open file model/train/BBT_emp.SpeakerDiarization.All.train/validate_SPEECH/BBT_emp.SpeakerDiarization.All.development/apply/0495/demofiles.SpeakerDiarization.All.test.rttm
 source line number 1
Feature extractors based on "yaafe" are not available because "yaafelib" could not be found.
Traceback (most recent call last):
  File "/anaconda3/envs/pyannote/bin/pyannote-multilabel", line 8, in <module>
    sys.exit(main())
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/pyannote/audio/applications/multilabel.py", line 463, in main
    _ = torch.Tensor([0]).to(device)
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 192, in _lazy_init
    _check_driver()
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 95, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
Extracting CHI
awk: can't open file model/train/BBT_emp.SpeakerDiarization.All.train/validate_SPEECH/BBT_emp.SpeakerDiarization.All.development/apply/0495/demofiles.SpeakerDiarization.All.test.rttm
 source line number 1
Feature extractors based on "yaafe" are not available because "yaafelib" could not be found.
Traceback (most recent call last):
  File "/anaconda3/envs/pyannote/bin/pyannote-multilabel", line 8, in <module>
    sys.exit(main())
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/pyannote/audio/applications/multilabel.py", line 463, in main
    _ = torch.Tensor([0]).to(device)
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 192, in _lazy_init
    _check_driver()
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 95, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
Extracting MAL
awk: can't open file model/train/BBT_emp.SpeakerDiarization.All.train/validate_SPEECH/BBT_emp.SpeakerDiarization.All.development/apply/0495/demofiles.SpeakerDiarization.All.test.rttm
 source line number 1
Feature extractors based on "yaafe" are not available because "yaafelib" could not be found.
Traceback (most recent call last):
  File "/anaconda3/envs/pyannote/bin/pyannote-multilabel", line 8, in <module>
    sys.exit(main())
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/pyannote/audio/applications/multilabel.py", line 463, in main
    _ = torch.Tensor([0]).to(device)
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 192, in _lazy_init
    _check_driver()
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 95, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
Extracting FEM
awk: can't open file model/train/BBT_emp.SpeakerDiarization.All.train/validate_SPEECH/BBT_emp.SpeakerDiarization.All.development/apply/0495/demofiles.SpeakerDiarization.All.test.rttm
 source line number 1
Feature extractors based on "yaafe" are not available because "yaafelib" could not be found.
Traceback (most recent call last):
  File "/anaconda3/envs/pyannote/bin/pyannote-multilabel", line 8, in <module>
    sys.exit(main())
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/pyannote/audio/applications/multilabel.py", line 463, in main
    _ = torch.Tensor([0]).to(device)
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 192, in _lazy_init
    _check_driver()
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 95, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
Extracting SPEECH
awk: can't open file model/train/BBT_emp.SpeakerDiarization.All.train/validate_SPEECH/BBT_emp.SpeakerDiarization.All.development/apply/0495/demofiles.SpeakerDiarization.All.test.rttm
 source line number 1
Feature extractors based on "yaafe" are not available because "yaafelib" could not be found.
Traceback (most recent call last):
  File "/anaconda3/envs/pyannote/bin/pyannote-multilabel", line 8, in <module>
    sys.exit(main())
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/pyannote/audio/applications/multilabel.py", line 463, in main
    _ = torch.Tensor([0]).to(device)
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 192, in _lazy_init
    _check_driver()
  File "/anaconda3/envs/pyannote/lib/python3.6/site-packages/torch/cuda/__init__.py", line 95, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
```

**Comment by MarvinLvn** [comment]
1) First, I've noticed that soxi wasn't installed on your configuration. I added it as a dependency.

```
brew install sox
```

should do the job !

2) `"yaafelib" could not be found`
This warning is thrown by pyannote, indicating that yaafelib being not found, you won't be able to extract yaafe features. As our model works on the waveform directly, this is not a problem. I might consider to catch it for better hiding it though :)

3) I've added an option to run the model on GPU. By default, it will run on CPU. I've modified the instructions accordingly. That should solve your problem with CUDA.

Let me know how things are going, and then I'll close the issue.
